### PR TITLE
Refactor journal entry acks.

### LINKED ISSUE #53: Allow any journal entries to require_ack
The `REQUIRES_ACK` flag is now restricted to custom journal entries, we should allow it for any entry.

With this mechanism in place, the SDK can decide to wait for acks of specific types of entries, to avoid losing side effects when dealing with the cancellation signal. See https://docs.google.com/document/d/14s7D6KP1IKNS1K3OrVbZNKGNei9TO_-9Sk0YC5mpht8/edit#heading=h.hooeo8f4d03z

I propose to keep this concept of "REQUIRES_ACK" general such that we can at a later point introduce more "cancellation points" without touching this aspect implementation in the runtime (and potentially in the sdks as well)

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fix #53 

* Generified the journal entry ack mechanism to any entry. Now the SDK implementation can ask the runtime to get acks for any entry.
* Created ad-hoc message for entries ack

### CODE REVIEW

File: dev/restate/service/protocol.proto [hunk 1]
```diff
@@ -53,6 +52,11 @@ message CompletionMessage {
   };
 }
 
+// Type: 0x0000 + 4
+message EntryAckMessage {
```
Comment (slinkydeveloper) [inline dev/restate/service/protocol.proto:56]: @tillrohrmann how do you feel about adding an ad hoc message for that?

Comment (tillrohrmann) [reply dev/restate/service/protocol.proto:56]: I guess this is fine as a start.

**Review by tillrohrmann — COMMENTED** [review]
The changes look good to me. Next time please give me a bit more time than 50 minutes to review the PR before you merge it.

File: dev/restate/service/protocol.proto (hunk 1 above)
Comment (slinkydeveloper) [reply dev/restate/service/protocol.proto:56]: The reason to add a new message is the following:

* First and foremost, the previous text here was wrong. we were not sending result.empty, but rather not sending result at all. The serialized protobuf field only had the entry index. this is something you can distinguish in the protobuf generated code, but it's not so nice.
* With the fact that we generalized this requires_ack mechanism, it seems to me a good-enough reason to give it an ad-hoc message.
* Fundamentally this makes little to no difference for the implementations.

Does that makes sense?

**Pull request merged**